### ISSUE TITLE
log: improve console channel format

### ISSUE THREAD

**Issue by x1B** [LaxarJS/laxar#445]
Currently, the console channel turns the `[1]`-style substitutions into a list of arguments to (e.g.) `console.log`. This leads to ugly additional spaces, making the output hard to read.

By now, pretty much all browsers should be supporting much of the [console standard](https://console.spec.whatwg.org/).

We could generate `%s` for string values, `%d` (or `%f`) for numbers and `%o` for everything else, hopefully creating a much more readable output.
Preliminary tests indicate that this improves on the spaces issue, at least in Chrome.

Also, I believe that the current substitution syntax does not correctly handle the case where a log message string _actually contains_ a standard placeholder such as `%o`, probably leading to unexpected results.

**Comment by jpommerening** [comment]
I gave it a shot and implemented an activity to show the effect:

```js
const object = { ... };
const array = [ ... ];
log.info( 'This is an object [0] and an array [1]', object, array );
```

Without my change, `console.info` is called like this:
```js
console.info(
   'INFO: ', 'log-demo-activity: This is an object ',
   object, ' and an array ', array,
   ' (widget-id: logDemoActivity-id0)', '(@ file:line)'
);
```

With my change it is:
```js
console.info(
   'INFO: log-demo-activity: This is an object %o and an array %o (widget-id: logDemoActivity-id0) (@ file:line)',
   object, array
);
```

## Results

### Chrome (before → after)
Chrome still does its flip-me-open-in-the-middle-of-a-sentence thing, but this time without a "preview" of the object contents.

![chrome-before](https://cloud.githubusercontent.com/assets/748636/24961204/6b61ab00-1f98-11e7-82ef-e1cf49b4fd12.jpg)
![chrome](https://cloud.githubusercontent.com/assets/748636/24961208/6b832b36-1f98-11e7-8c66-d1057bbc37a0.jpg)

### Firefox (before → after)
Firefox still puts the trailing string in quotes, even if it's not a separate argument! (Try `console.log( 'This is a %s test', 'fine' )`)

![firefox-before](https://cloud.githubusercontent.com/assets/748636/24961207/6b828e74-1f98-11e7-87b9-d7d0e43415eb.jpg)
![firefox](https://cloud.githubusercontent.com/assets/748636/24961209/6b83953a-1f98-11e7-92bc-6f73f8fef2a5.jpg)

### Safari (before → after)
Visible improvements here. Safari distinguishes visually between passing multiple args (rendered as a list, including quotes indicating string arguments, and passing a format string plus replacements.

![safari-before](https://cloud.githubusercontent.com/assets/748636/24961205/6b7e0a02-1f98-11e7-96e5-9286a811b859.jpg)
![safari](https://cloud.githubusercontent.com/assets/748636/24961206/6b818bc8-1f98-11e7-86fe-654f02103791.jpg)

## Putting [💄](https://www.npmjs.com/package/color-rainbow) on the [🐘](https://developer.mozilla.org/en-US/docs/Web/API/console#Styling_console_output) in the room
![safari-rainbow](https://cloud.githubusercontent.com/assets/748636/24963293/7bed725a-1f9e-11e7-8f07-e7b4b69d14b6.jpg)
Ok, maybe we should not go _that_ far

**Comment by x1B** [comment]
Yes, very promising. Also not surprised about your results with Firefox. What about plain Strings? I think they annoyed me the most.

From reading the diff, this
```js
log.info( 'This is a "[0]"', 'string' );
```
 should now read fluently like this:
```
console.info( 'This is a "%s"', 'string' )
```
Right?

Re 💄: how on earth...?

**Comment by jpommerening** [comment]
Yes, it should! And it does! Strings and numbers just "disappear" into the log message. And it's all `%c`ss!

**Comment by jpommerening** [comment]
Referring to the documentation of [`utilities.string`](https://github.com/LaxarJS/laxar/blob/master/docs/api/utilities.string.md#module-members), I noticed how this is pretty much compatible with the console spec.

I'll update the parsing to use the defaults as described above (strings `%s`, numbers `%f`, everything else `%o`) but will use the formatting specifier if a specifier was given that is compatible with the console spec.

**Comment by jpommerening** [comment]
Implemented on `master` (`v2.0.0`)
